sync-bitcoin.js

```diff
@@ -0,0 +1,72 @@
+const { closeBudget, openBudget, getTransactions, getAccountNote, getAccountBalance, ensurePayee } = require('./utils');
+const api = require('@actual-app/api');
+
+function getValueAtPath(obj, path) {
+    const keys = path.split('.').filter(Boolean);
+    
+    return keys.reduce((acc, key) => {
+        const match = key.match(/^([^\[\]]+)(\[(\d+)\])?$/);
+        
+        if (match) {
+            const property = match[1];
+            const index = match[3];
+
+            acc = acc[property];
+
+            if (index !== undefined) {
+                acc = acc[parseInt(index, 10)];
+            }
+        } else {
+            acc = acc[key];
+        }
+        
+        return acc;
+    }, obj);
+}
+
+async function getBitcoinPrice() {
+  let response = undefined;
+  const url = process.env.BITCOIN_PRICE_URL || "https://api.kraken.com/0/public/Ticker?pair=xbtusd"
+  const path = process.env.BITCOIN_PRICE_JSON_PATH || "result.XXBTZUSD.c[0]"
+  try {
+    response = await fetch(url);
+    const json = await response.json();
+    return getValueAtPath(json, path);
+  } catch (error) {
+    return undefined;
+  }
+}
+
+(async () => {
+  await openBudget();
+  const bitcoinPrice = await getBitcoinPrice();
+  if (!bitcoinPrice) {
+    throw new Error("Unable to retrieve Bitcoin price. Check your BITCOIN_PRICE_URL and BITCOIN_PRICE_JSON_PATH environment variables");
+  }
```

Comment:
Do this after getting the bc price (so if that fails, the budget file was not opened).

```suggestion
  const bitcoinPrice = await getBitcoinPrice();
  if (!bitcoinPrice) {
    throw new Error("Unable to retrieve Bitcoin price. Check your BITCOIN_PRICE_URL and BITCOIN_PRICE_JSON_PATH environment variables");
  }
  await openBudget();
```